### PR TITLE
nspawn controlling tty tweaks

### PR DESCRIPTION
**Pull request by poettering**
Nothing too earth shattering. Prompted by #17070

### CODE REVIEW

**Comment by snarkmaster** [comment]
@poettering, in view of my update on #17070, maybe it is clearer why I use `--console=pipe` in interactive mode. As such, my customers will see your `log_notice` on every invocation. Can you please make it fit on one line?

Also, if we're concerned about users' security postures, shouldn't we also check `isatty(STDERR_FILENO)`? 

Here is a suggestion of an 80-char message, assuming you check all 3 FDs.

```
Insecure container: `--console=pipe` while one of std{in,out,err} is a TTY.
```

File: src/nspawn/nspawn.c [hunk 1]
```diff
@@ -275,9 +277,14 @@ static int handle_arg_console(const char *arg) {
                 arg_console_mode = CONSOLE_READ_ONLY;
         else if (streq(arg, "passive"))
                 arg_console_mode = CONSOLE_PASSIVE;
-        else if (streq(arg, "pipe"))
+        else if (streq(arg, "pipe")) {
+                if (isatty(STDIN_FILENO) > 0 && isatty(STDOUT_FILENO) > 0)
+                        log_notice("Console mode 'pipe' selected, but standard input/output are connected to an interactive TTY. "
```
Comment (anitazha) [inline src/nspawn/nspawn.c]: Downgrade to debug on quiet?

Comment (poettering) [reply src/nspawn/nspawn.c]: yupp that makes sense!

**Comment by poettering** [comment]
@snarkmaster i added a commit that adds `--console=autopipe`, wouldn't that be what you are looking for?

**Comment by poettering** [comment]
force pushed a new version, only two changes are @anitazha's suggested `--quiet` logic and the autopipe thing.

**Comment by anitazha** [comment]
These changes LGTM. bionic timed out like it tends to do.

@snarkmaster did you have any additional comments with autopipe before merging?

**Pull request merged**